backend/app/store/search/internal/bleve.go

```diff
@@ -0,0 +1,235 @@
+package internal
+
+import (
+	"encoding/hex"
+	"fmt"
+	"hash/fnv"
+	"os"
+	"path"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+
+	"github.com/blevesearch/bleve"
+	bleveCustom "github.com/blevesearch/bleve/analysis/analyzer/custom"
+	bleveStandard "github.com/blevesearch/bleve/analysis/analyzer/standard"
+	bleveEn "github.com/blevesearch/bleve/analysis/lang/en"
+	bleveRu "github.com/blevesearch/bleve/analysis/lang/ru"
+
+	bleveSingle "github.com/blevesearch/bleve/analysis/tokenizer/single"
+
+	"github.com/blevesearch/bleve/analysis/token/lowercase"
+	"github.com/blevesearch/bleve/mapping"
+	"github.com/pkg/errors"
+
+	service "github.com/umputun/remark42/backend/app/store/search/service"
+)
+
+const urlFieldName = "url"
+const textFieldName = "text"
+
+// Available text analyzers.
+// Bleve supports a bit more languages that may be added,
+// see https://github.com/blevesearch/bleve/tree/master/analysis/lang
+var analyzerMapping = map[string]string{
+	"standard": bleveStandard.Name,
+	"english":  bleveEn.AnalyzerName,
+	"russian":  bleveRu.AnalyzerName,
+}
+
+type bleveBatch struct {
+	*bleve.Batch
+}
+
+type bleveIndexer struct {
+	bleve.Index
+}
+
+// Index adds document to current batch
+func (b bleveBatch) Index(id string, data *service.DocumentComment) error {
+	return b.Batch.Index(id, data)
+}
+
+func newBleve(indexPath, analyzer string) (s *bufferedEngine, err error) {
+	if _, ok := analyzerMapping[analyzer]; !ok {
+		analyzers := make([]string, 0, len(analyzerMapping))
+		for k := range analyzerMapping {
+			analyzers = append(analyzers, k)
+		}
+		return nil, errors.Errorf("Unknown analyzer: %q. Available analyzers for bleve: %v", analyzer, analyzers)
+	}
+	var index bleve.Index
+
+	st, errOpen := os.Stat(indexPath)
+	switch {
+	case os.IsNotExist(errOpen):
+		// create new
+		log.Printf("[INFO] creating new search index %s", indexPath)
+		index, err = bleve.New(indexPath, createIndexMapping(analyzerMapping[analyzer]))
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot open index")
+		}
+	case errOpen == nil:
+		// open existing
+		if !st.IsDir() {
+			return nil, errors.Errorf("index path should be a directory")
+		}
+		log.Printf("[INFO] opening existing search index %s", indexPath)
+		index, err = bleve.Open(indexPath)
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot create index")
+		}
+	default:
+		// error
+		return nil, errors.Wrap(err, "cannot open index")
+	}
+
+	eng := &bufferedEngine{
+		index:         bleveIndexer{index},
+		queueNotifier: make(chan bool),
+		flushEvery:    2 * time.Second,
+		flushCount:    100,
+		indexPath:     indexPath,
+	}
+
+	eng.shutdownWait.Add(1)
+	// exit when queueNotifier is closed and decrements shutdownWait counter
+	go eng.indexDocumentWorker()
+
+	return eng, nil
+}
+
+// NewBleveService create search sservice based on bleve engine
+func NewBleveService(params service.SearcherParams) (s service.Service, err error) {
```

Comment:
import cycle. You should return `Bleve` object which signature satisfies `service.Service`, and have no mentions of `service.*` in internal module to fix that.